implementations/micrometer-registry-cloudwatch2/src/main/java/io/micrometer/cloudwatch2/CloudWatchMeterRegistry.java

```diff
@@ -295,6 +296,10 @@ StandardUnit toStandardUnit(@Nullable String unit) {
 
         private List<Dimension> toDimensions(List<Tag> tags) {
             return tags.stream()
+                    // Take restrictions for dimensions in CloudWatch into account
+                    // https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_Dimension.html
+                    .filter(t -> StringUtils.isNotBlank(t.getValue()))
+                    .filter(t -> t.getValue().length() <= 255)
```

Comment:
For maximum length constraints, we have historically truncated the tag value. We do this in a `NamingConvention` implementation that the MeterRegistry is configured to use.

Reply:
Done.